quote-server/server.js

```diff
@@ -1,32 +1,34 @@
-// server.js
-// This is where your node app starts
-
-//load the 'express' module which makes writing webservers easy
 import express from "express";
-//load the quotes JSON
-import quotes from "./quotes.json" assert { type: "json" };
+import fetch from "node-fetch";
+import cors from "cors";
 
 const app = express();
-// Now register handlers for some routes:
-//   /                  - Return some helpful welcome info (text)
-//   /quotes            - Should return all quotes (json)
-//   /quotes/random     - Should return ONE quote (json)
-app.get("/", (request, response) => {
-  response.send("Neill's Quote Server!  Ask me for /quotes/random, or /quotes");
-});
 
-//START OF YOUR CODE...
 
-//...END OF YOUR CODE
+app.use(cors());
+
+const getQuotes = async (url) => {
+  const response = await fetch(url);
+  return response.json();
+};
+
+app.get("/", (req, res) => {
+  res.send("Welcome to Fathi's Quote Server! Use /quotes/random or /quotes to get quotes.");
+});
+
+app.get("/quotes", async (req, res) => {
+  const pageNumber = req.query.page || 1;
+  const allQuotes = await getQuotes(`https://api.quotable.io/quotes?page=${pageNumber}`);
+  res.send(allQuotes);
+});
 
-//You can use this function to pick one element at random from a given array
-//example: pickFromArray([1,2,3,4]), or
-//example: pickFromArray(myContactsArray)
-//
-const pickFromArray = (arrayofQuotes) =>
-  arrayofQuotes[Math.floor(Math.random() * arrayofQuotes.length)];
+app.get("/quotes/random", async (req, res) => {
+  const randomQuoteResponse = await fetch("https://api.quotable.io/random");
+  const randomQuote = await randomQuoteResponse.json();
+  res.send(randomQuote);
+});
```

Comment:
Good job!

One question, could you reuse the `getQuotes` method for both the `/quotes` and the `/quotes/random` controller?

Reply:
Well done finding a simplified solution, but did you try to use the pickFromArray method provided before arriving at this solution?

Reply:
> Good job!
> 
> One question, could you reuse the `getQuotes` method for both the `/quotes` and the `/quotes/random` controller?

Yes  indeed, thanks, it will definitely reduce code duplication.